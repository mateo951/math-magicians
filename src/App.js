import React from 'react';
import Calculator from './components/Calculator';

function App() {
  return (
    <div className="App">
      <div className="container">
        <h1>Hello App!</h1>
        <Calculator />
      </div>
    </div>
  );
}

export default App;
