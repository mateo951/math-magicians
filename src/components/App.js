import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Navigation from './Navigation';
import Home from './Home';
import Calculator from './Calculator';
import Quotes from './Quotes';

const App = () => (
  <>
    <Navigation />
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>
      <Route path="/calculator">
        <Calculator />
      </Route>
      <Route path="/quotes">
        <Quotes />
      </Route>
    </Switch>
  </>
);

export default App;
