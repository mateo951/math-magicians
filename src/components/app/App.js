import React from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import Navigation from '../navigation/Navigation';
import Home from '../home/Home';
import Calculator from '../calculator/Calculator';
import Quotes from '../quotes/Quotes';

const App = () => (
  <Router>
    <Navigation />
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>
      <Route path="/calculator">
        <Calculator />
      </Route>
      <Route path="/quotes">
        <Quotes />
      </Route>
    </Switch>
  </Router>
);

export default App;
