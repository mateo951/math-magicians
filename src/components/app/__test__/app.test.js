import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import App from '../App';

describe('pages test snapshots', () => {
  it('testing home page', () => {
    render(<App />);
    const Home = screen.getByText('Home');
    fireEvent.click(Home);
    expect(screen.getByText('Welcome to our page')).toMatchSnapshot();
  });
  it('testing Quote page', () => {
    render(<App />);
    const Quote = screen.getByText('Quotes');
    fireEvent.click(Quote);
    expect(screen.getByText('A language that does not affect the way you think about programming is not worth knowing.')).toMatchSnapshot();
  });
  it('testing Calculator page', () => {
    render(<App />);
    const Calculator = screen.getByText('Calculator');
    fireEvent.click(Calculator);
    expect(screen.getByText('0')).toMatchSnapshot();
  });
});