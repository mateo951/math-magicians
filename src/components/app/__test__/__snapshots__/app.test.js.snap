// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages test snapshots testing Calculator page 1`] = `
<input
  class="calcItem cero"
  type="button"
  value="0"
/>
`;

exports[`pages test snapshots testing Quote page 1`] = `
<h1
  class="quote"
>
  A language that does not affect the way you think about programming is not worth knowing.
</h1>
`;

exports[`pages test snapshots testing home page 1`] = `
<h1
  class="title"
>
  Welcome to our page
</h1>
`;
